File: app/src/main/java/ru/magflayer/spectrum/domain/repository/PhotoRepository.java
package ru.magflayer.spectrum.domain.repository;

import java.util.List;

import ru.magflayer.spectrum.domain.entity.ColorPhotoEntity;
import rx.Observable;

public interface PhotoRepository {

    Observable<Boolean> savePhoto(ColorPhotoEntity colorPhoto);

    Observable<List<ColorPhotoEntity>> loadPhotos();

    Observable<ColorPhotoEntity> loadPhoto(String filePath);

    Observable<Boolean> removePhoto(ColorPhotoEntity entity);

}

File: app/src/main/java/ru/magflayer/spectrum/domain/repository/PhotoRepository.kt
package ru.magflayer.spectrum.domain.repository

import ru.magflayer.spectrum.domain.entity.ColorPhotoEntity

interface PhotoRepository {

    suspend fun savePhoto(colorPhoto: ColorPhotoEntity): Boolean

    suspend fun loadPhotos(): List<ColorPhotoEntity>

    suspend fun loadPhoto(filePath: String): ColorPhotoEntity

    suspend fun removePhoto(entity: ColorPhotoEntity): Boolean
}
